pilot/pkg/xds/workload.go

```diff
@@ -56,7 +56,7 @@ func (e WorkloadGenerator) GenerateDeltas(
 		return e.generateDeltasOndemand(proxy, req, w)
 	}
 
-	subs := w.ResourceNames
+	subs := req.Delta.Subscribed
```

Comment:
I'm curious: do we use the subscribed list in non-WDS codepaths?

Reply:
Yeah we actually do basically this exact thing here: https://github.com/istio/istio/blob/d14a7d7b2805dfbb814873d4861b39714c093a75/pilot/pkg/xds/delta.go#L487. In theory we could do the same here but we only do this for non-on-demand. Ill make a reference to it in a comment